lib/ansible/module_utils/urls.py

```diff
@@ -302,6 +302,16 @@ def match_hostname(cert, hostname):
 
     HAS_MATCH_HOSTNAME = True
 
+if hasattr(socket, 'create_connection'):
+    create_connection = socket.create_connection
+else:
+    # Just a graceful fallback for py < 2.6, implementing old behaviour.
+    # Wouldn't work with IPv6, but it is unlikely that python 2.4 could be met
+    # on IPv6-only machine.
+    def create_connection(address):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(address)
+        return sock
```

Comment:
I don't create a function named the same as original method, I just use it if it is available, and define a fallback if it is not available (in which case there is no original method). For python>2.4 this function would be the original method. There is no confusion.

The problem with `hasattr` is not a problem since we use it not on a custom class instance but on a module.

Also, it doesn't perform lookup multiple times, since it is called on module import.